### DISCUSSION TITLE
Throughput Statistics from the Steady Lads Mint of June 2022

### DISCUSSION THREAD

**Discussion by gregorycoppola** [stacks-network/stacks-core#3203] [Ideas]
# Abstract

On June 29, 2022, there was a relatively large NFT mint on the Stacks chain. This was *Megapont*’s 10k NFT [Steady Lads](https://twitter.com/MegapontNFT/status/1541385974655107073) mint.

This is an interesting event because it is a relatively large stress test of the Stacks network.

This discussion looks at some statistics from the launch.

# Throughput vs. Mempool Visualization

![mempool-throughput](https://user-images.githubusercontent.com/60008382/179428150-89ae46c7-f199-48c5-b071-6a8885121390.png)

We are looking at a graph of **transactions per block** versus **mempool size**.

There has previously been debate about whether unfilled blocks are only a result of there not being any work to do on them in the mempool. From this graph, we can see that:

- there are **many single-digit blocks**, even **during the middle of the launch**, with multi-thousands of transactions in the mempool
- **some** **miners** manage to get relatively **full** **blocks** of hundreds of transactions, even up to 1000 transactions per block
  - but, at the same time, **other miners** are producing **nearly empty blocks**
- the mint of **10k NFT’s** took around **3 days** to clear

# Throughput Statistics

We calculated **throughput** **statistics** over the following roughly **three-day period**, comprising what looked like the bulk of the mint:

- **start** `2022-06-29T20:29:00.000Z`
- **end** `2022-07-02T16:40:51.000Z`

We made statistics for the following measures:

- **transactions per second**
  - amortized over the past **6 Stacks blocks**
  - compares **total** **transactions** to **time** **elapsed**
- **transactions per block**
  - this information is part of the **block** **itself**
- **block fullness**
  - the **percentage** (out of 100%) of the **block** that was **used**
    - measured as the percentage of the **read count** used
    - read count is basically always the **bottleneck**

We observed the following data:

|             | transactions / second | transactions / block | percent full |
| ----------- | --------------------- | -------------------- | ------------ |
| **average** | 0.108                 | 76.5                 | 19.1%        |
| **maximum** | 0.944                 | 1054                 | 100%         |

# Underlying Data

This is based on the following raw data:

- **mempool**
  - https://raw.githubusercontent.com/gregorycoppola/stx-share/main/data/steady-lads-1-mempool.raw
- **blocks**
  - https://raw.githubusercontent.com/gregorycoppola/stx-share/main/scrape-blocks/output/2022jul17

**Comment by wileyj** [comment]
@jcnelson @kantai thoughts?

**Reply by wileyj** [reply]
@obycode

**Comment by obycode** [comment]
It looks like this mempool data collected is just a count of the number of transactions in the mempool. It would be great if we collected snapshots of all of the transactions in the mempool, so that experiments could be done to evaluate different methods for selecting and mining.
